### PR TITLE
Fixes #17910: Error during translate in CIS techniques

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17910

### CODE REVIEW

**Comment by gpoblon** [comment]
I still have issues with it, not ready to merge

File: rudder-lang/src/translate.rs [hunk 1]
```diff
@@ -319,8 +319,8 @@ resource {bundle_name}({parameter_list})
 
     fn translate_class(&self, cond: &str) -> Result<String> {
         lazy_static! {
-        static ref METHOD_RE: Regex = Regex::new(r"^(\w+)_(\w+)$").unwrap();
-    }
+            static ref CLASS_RE: Regex = Regex::new(r"(?U)^([\w${.}]+)(?:_(not_repaired|repaired|false|true|not_ok|ok|reached|error|failed|denied|timeout|success|not_kept|kept))?$").unwrap();
```
Comment (amousset) [inline rudder-lang/src/translate.rs:322]: classes are now named conditions in rudder

File: rudder-lang/src/translate.rs [hunk 2]
```diff
@@ -338,28 +338,27 @@ resource {bundle_name}({parameter_list})
...
+            let outcome = match caps.get(2) {
+                Some(res) => res.as_str(),
+                None => return Ok(method.to_owned())
+            };
+            if vec!["kept", "success"].iter().any(|x| x == &outcome) {
                 return Ok(format!("{} =~ success", method));
-            } else if vec!["error", "not_ok", "failed", "denied", "timeout"]
-                .iter()
-                .any(|x| x == &status)
+            } else if vec!["error", "not_ok", "failed", "denied", "timeout", "not_kept"].iter().any(|x| x == &outcome)
```
Comment (amousset) [inline rudder-lang/src/translate.rs:355]: not_kept means error or repaired

**Comment by gpoblon** [comment]
OK, merging this PR

**Pull request merged**